include/app/config.hpp

```diff
@@ -111,6 +111,7 @@ class Config : public QObject {
 
     std::shared_ptr<f1x::openauto::autoapp::configuration::Configuration> openauto_config;
 
+    inline QSettings *get_settings() { return &this->ia_config; };
```

Comment:
i dont think we should do this

i know it probably isn't necessary to have methods defined for every config option, but the QSettings object should really be internal to the class

Reply:
Totally agree. The only reason I included this was that I wanted to prove it was possible to use settings in a clear and principled way without a dedicated object that combines the concerns of many different code components.

Reply:
I can move my code into the Config object but I would suggest we consider another way. Config  object isn't very nice.

Reply:
Yeah i really didnt expect so many config options when I started doing it that way so it wasn't that bad haha but as i started adding more and more and then in just got to where it is now lol

what other code components are in the config tho?

Reply:
Heh, I must have been looking at something else. The config object seems to be just a bunch of getters and setters. But that is also what QSettings is in the first place. I don't see the value.

Reply:
well the config class defines types for every option, so you don't have to worry about what type you need to get out of the variant

also helps to not have the string option keys all throughout the code

Reply:
Typing is good. Though the strings aren't scattered throughout the code, the function names are so unless you create a mismatch between the ini and the code you have to change them everywhere either way.

Reply:
Well imagine you have a setting option "A" that needs to be used by 2 functions. If you allow the settings to be accessed directly then the 2nd function could possibly typo it as "a" and thus will be a different option.

With the "wrapper" class, if there's a typo you'll know when you compile it :p